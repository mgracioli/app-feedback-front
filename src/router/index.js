import { createRouter, createWebHistory } from 'vue-router'

const Home = () => import('../views/Home')
const Feedbacks = () => import('../views/Feedbacks')
const Credentials = () => import('../views/Credentials')

const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: Feedbacks,
    meta: {
      hasAuth: true // para acessar essa rota, é necesário estar autenticado
    }
  },
  {
    path: '/credentials',
    name: 'Credentials',
    component: Credentials,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/:pathMatch(.*)*', // regex do Vue Router (caso não encontre a rota nas rotas definidas acima, redireciona para a Home)
    redirect: 'Home'
  }
]

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

export default router
