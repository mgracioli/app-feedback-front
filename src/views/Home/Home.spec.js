import Home from '.'
import { routes } from '../../router'
import { createRouter, createWebHistory } from 'vue-router'
import { shallowMount } from '@vue/test-utils'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

describe('views.Home', () => {
  it('Deve carregar a página Home corretamente', async () => {
    router.push('/')
    await router.isReady()  //os testes só serão rodados quando o router estiver pronto

    //shallowMount - monta só o componente Home, não monta os componentes internos da Home (ex.: CustomHeader, Contact...)
    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router]
      }
    })

    //toMatchSnapShot vai criar um snapshot, se ele ainda não existir, e comparar com o que está sendo carregado, se ambos forem iguais, oteste passa (nesse caso, ele vai serializar o html  da página Home, ou seja, transformar seu HTML em uma string e comparar ele com o HTML gerado nas proximas vezes que o teste for rodado)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
