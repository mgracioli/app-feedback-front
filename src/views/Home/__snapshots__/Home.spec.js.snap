// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views.Home Deve carregar a página Home corretamente 1`] = `
<custom-header-stub></custom-header-stub>
<contact-stub></contact-stub>
<div class="flex justify-center py-10 bg-brand-gray">
  <p class="font-medium text-center text-gray-800">App Feedback © 2023</p>
</div>
`;
